File: 25200/BOJ_25214.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int min = Integer.MAX_VALUE;
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        int ans = 0;
        while (n-->0) {
            int cur = Integer.parseInt(st.nextToken());
            if (min > cur) min = cur;
            else ans = Math.max(ans, cur-min);
            sb.append(ans).append(' ');
        }
        System.out.println(sb);
    }
    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 25200/BOJ_25214.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    var min = Int.MAX_VALUE
    val st = StringTokenizer(readLine())
    val sb = StringBuilder()
    var ans = 0
    repeat(n) {
        val cur = st.nextToken().toInt()
        if (min > cur) min = cur
        else ans = Math.max(ans, cur-min)
        sb.append(ans).append(' ')
    }
    println(sb)
}
